cve_bin_tool/extractor.py

```diff
@@ -103,16 +112,19 @@ def extract_file_deb(cls, filename, extraction_path):
     @classmethod
     def extract_file_cab(cls, filename, extraction_path):
         """ Extract cab files """
-        if not inpath("cabextract"):
-            print("Error: need 'cabextract' binary in path")
+        if sys.platform == "linux":
+            if not inpath("cabextract"):
+                raise Exception("'cabextract' is required to extract cab files")
+            else:
+                return subprocess.call(["cabextract", "-d", extraction_path, filename])
         else:
-            return subprocess.call(["cabextract", "-d", extraction_path, filename])
+            subprocess.call("Expand " + filename + " -F:* " + extraction_path)
```

Comment:
Should this be subprocess.call(["Expand", filename, "-F:*", extraction_path]) or does that not work on windows?

Reply:
Explanation: subprocess does input validation for you and it's considered better practice to split things into individual arguments so it can do the right thing.   It probably doesn't make any difference for well-formed filenames, but this all-in-one-string might behave oddly with unusual characters in the filename or extraction path.  I would *assume* that's true on windows too, but I've never verified it myself.